pkg/resources/network_policy.go

```diff
@@ -20,7 +24,6 @@ var networkPolicySchema = map[string]*schema.Schema{
 		Type:        schema.TypeString,
 		Required:    true,
 		Description: "Specifies the identifier for the network policy; must be unique for the account in which the network policy is created.",
-		ForceNew:    true,
 	},
 	"allowed_network_rule_list": {
```

Comment:
are any suppressions needed for network rule lists? (https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2236#issuecomment-2220003509)

Reply:
Added suppress diff that should prevent this, will be useful in the future as lists/sets of identifiers are pretty common.

Reply:
Left a few comments in tests. And this diff suppressor looks a bit like magic :( But looks fine (functionally)

Reply:
Yeah, unfortunately, it is magical, but I couldn't find any better way of addressing suppressions on Sets that are painful to work with (+ additional pain from identifiers).